app/jobs/scheduled/clean_up_inactive_users.rb

```diff
@@ -0,0 +1,28 @@
+module Jobs
+
+  class CleanUpInactiveUsers < Jobs::Scheduled
+    every 1.day
+
+    def execute(args)
+      destroyer = UserDestroyer.new(Discourse.system_user)
+
+      User.joins("LEFT JOIN posts ON posts.user_id = users.id")
+        .where("posts.user_id IS NULL")
+        .where("users.last_seen_at < ?", 2.years.ago)
+        .where(trust_level: 0)
```

Comment:
I think this is probably ok, maybe add `where(last_posted_at: nil)` to speed up the query a tiny bit (you can still keep the join)

Reply:
`2.years.ago` should be a site setting, some sites may want this to be more aggressive than 2 years, others may want to disable it.